bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/api/EcobeeAuth.java

```diff
@@ -0,0 +1,269 @@
+            case NEED_TOKEN:
+                getTokens();
+                break;
+            case COMPLETE:
+                bridgeHandler.updateBridgeStatus(ThingStatus.ONLINE);
+                break;
+        }
+        return state;
+    }
+
+    /**
+     * Call the Ecobee authorize endpoint to get the authorization code and PIN
+     * that will be used a) validate the application in the the Ecobee user web portal,
+     * and b) make the first time request for the access and refresh tokens.
+     * Warnings are suppressed to avoid the Gson.fromJson warnings.
+     */
+    @SuppressWarnings({ "null", "unused" })
+    private void authorize() throws EcobeeAuthException {
+        logger.debug("EcobeeAuth: State is {}: Executing step: 'authorize'", state);
+        StringBuilder url = new StringBuilder(ECOBEE_AUTHORIZE_URL);
+        url.append("?response_type=ecobeePin");
+        url.append("&client_id=").append(apiKey);
+        url.append("&scope=").append(ECOBEE_SCOPE);
+
+        logger.trace("EcobeeAuth: Getting authorize URL={}", url.toString());
+        String response = executeUrl("GET", url.toString());
+        logger.trace("EcobeeAuth: Auth response: {}", response);
+
+        try {
+            authResponse = EcobeeApi.getGson().fromJson(response, AuthorizeResponseDTO.class);
+            if (authResponse == null) {
+                logger.debug("EcobeeAuth: Got null authorize response from Ecobee API");
+                setState(EcobeeAuthState.NEED_PIN);
+            } else {
+                if (StringUtils.isNotEmpty(authResponse.error)) {
+                    throw new EcobeeAuthException(authResponse.error + ": " + authResponse.errorDescription);
+                }
+                code = authResponse.code;
+                writeLogMessage(authResponse.pin, authResponse.expiresIn);
+                setPinExpirationTime(authResponse.expiresIn.longValue());
+                updateBridgeStatus();
+                setState(EcobeeAuthState.NEED_TOKEN);
+            }
+        } catch (JsonSyntaxException e) {
+            logger.info("EcobeeAuth: Exception while parsing authorize response: {}", e.getMessage());
+            setState(EcobeeAuthState.NEED_PIN);
+        }
+    }
+
+    /**
+     * Call the Ecobee token endpoint to get the access and refresh tokens. Once successfully retrieved,
+     * the access and refresh tokens will be injected into the OHC OAuth service.
+     * Warnings are suppressed to avoid the Gson.fromJson warnings.
+     */
+    @SuppressWarnings({ "null", "unused" })
+    private void getTokens() throws EcobeeAuthException {
+        logger.debug("EcobeeAuth: State is {}: Executing step: 'getToken'", state);
+        StringBuilder url = new StringBuilder(ECOBEE_TOKEN_URL);
+        url.append("?grant_type=ecobeePin");
+        url.append("&code=").append(code);
+        url.append("&client_id=").append(apiKey);
+
+        logger.trace("EcobeeAuth: Posting token URL={}", url.toString());
```

Comment:
toString is handled for you by the logger.
```suggestion
        logger.trace("EcobeeAuth: Posting token URL={}", url);
```

Reply:
Fixed